source/iterator/cdc.go

```diff
@@ -0,0 +1,487 @@
+	var err error
+
+	it := &cdcIterator{
+		db:            params.db,
+		table:         params.table,
+		trackingTable: params.trackingTable,
+		keys:          params.keys,
+		batchSize:     params.batchSize,
+		position:      params.position,
+		columnTypes:   params.columnTypes,
+		tableSrv:      newTrackingTableService(),
+	}
+
+	if err = it.loadRows(ctx); err != nil {
+		return nil, fmt.Errorf("load rows: %w", err)
+	}
+
+	// run clearing tracking table.
+	go it.clearTrackingTable(ctx)
+
+	return it, nil
+}
+
+// HasNext check ability to get next record.
+//
+//nolint:funlen,nolintlint
+func (i *cdcIterator) HasNext(ctx context.Context) (bool, error) {
+	if i.rows != nil && i.rows.Next() {
+		return true, nil
+	}
+
+	if err := i.loadRows(ctx); err != nil {
+		return false, fmt.Errorf("load rows: %w", err)
+	}
+
+	return false, nil
+}
+
+// Next get new record.
+// nolint:funlen,nolintlint
+func (i *cdcIterator) Next(ctx context.Context) (sdk.Record, error) {
+	row := make(map[string]any)
+	if err := i.rows.MapScan(row); err != nil {
+		return sdk.Record{}, fmt.Errorf("scan rows: %w", err)
+	}
+
+	transformedRow, err := columntypes.TransformRow(ctx, row, i.columnTypes)
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("transform row column types: %w", err)
+	}
+
+	id, ok := transformedRow[columnTrackingID].(int64)
+	if !ok {
+		return sdk.Record{}, ErrWrongTrackingIDType
+	}
+
+	operationTypeBt, ok := transformedRow[columnOperationType].([]byte)
+	if !ok {
+		return sdk.Record{}, ErrWrongTrackingOperatorType
+	}
+
+	pos := position.Position{
+		IteratorType: position.TypeCDC,
+		CDCLastID:    int(id),
+		SuffixName:   i.trackingTable[len(i.trackingTable)-6:],
+	}
+
+	convertedPosition, err := pos.ConvertToSDKPosition()
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("convert position %w", err)
+	}
+
+	keysMap := make(map[string]any)
+	for _, val := range i.keys {
+		if _, ok := transformedRow[val]; !ok {
+			return sdk.Record{}, fmt.Errorf("key %v, %w", val, ErrNoKey)
+		}
+		keysMap[val] = transformedRow[val]
+	}
+
+	delete(transformedRow, columnOperationType)
+	delete(transformedRow, columnTrackingID)
+
+	transformedRowBytes, err := json.Marshal(transformedRow)
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("marshal row: %w", err)
+	}
+
+	i.position = &pos
+	metadata := sdk.Metadata(map[string]string{metadataTable: i.table})
+	metadata.SetCreatedAt(time.Now())
+
+	switch actionType(operationTypeBt) {
+	case insertOperation:
+		return sdk.Util.Source.NewRecordCreate(convertedPosition, metadata,
+			sdk.StructuredData(keysMap), sdk.RawData(transformedRowBytes)), nil
+	case updateOperation:
+		return sdk.Util.Source.NewRecordUpdate(convertedPosition, metadata,
+			sdk.StructuredData(keysMap), nil, sdk.RawData(transformedRowBytes)), nil
+	case deleteOperation:
+		return sdk.Util.Source.NewRecordDelete(convertedPosition, metadata,
+			sdk.StructuredData(keysMap)), nil
+	default:
+		return sdk.Record{}, ErrUnknownOperatorType
+	}
```

Comment:
Same as in other places: IMHO it's more natural that the payload for DB sources comes in the form of `sdk.StructuredData`.

Reply:
please check it
https://github.com/conduitio-labs/conduit-connector-sap-hana/pull/12#discussion_r1097502451